src/app/youtube/pages/preview/preview.component.ts

```diff
@@ -0,0 +1,38 @@
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import SearchItem from '../../models/search-item.model';
+import ItemService from '../../service/item/item.service';
+
+@Component({
+  selector: 'app-preview',
+  templateUrl: './preview.component.html',
+  styleUrls: ['./preview.component.scss'],
+})
+class PreviewComponent implements OnInit {
+  isLoading : boolean = false;
+
+  cardInfo! : SearchItem;
+
+  constructor(
+    private itemService : ItemService,
+    private activatedRoute: ActivatedRoute,
+    private router : Router,
+  ) {}
+
+  ngOnInit(): void {
+    this.isLoading = true;
+    const id = this.activatedRoute.snapshot.url[0].path;
+    this.itemService.getSingleItem(id).subscribe({
```

Comment:
`take` or `takeUntil` needed for all subscriptions, since otherwise you are leaving subscription forever opened which will introduce strange behaviours and memory leaks